src/docs_website/src/html/404.html

```diff
@@ -0,0 +1,225 @@
+        { x: bugx + 2, y: bugy + 12, w: 32, h: 28 };
+      if (gamestate == STATE_GAME && overlap(playerbox, bugbox)) {
+        gamestate = STATE_OVER;
+        gameovertime = time;
+      }
+
+      let playerframe = int(time * 10) % 4;
+      if (gamestate == STATE_GAME) {
+        playerframe = int(time * speed / 20) % 4 + 4;
+        if (jumping) playerframe = playerdy > 0 ? 4 : playery > 60 ? 5 : 6;
+      } else if (gamestate == STATE_OVER) {
+        playerframe += 10;
+        const hurttime = time - gameovertime;
+        if (hurttime < 0.3 || jumping) playerframe = int(hurttime * 10) % 2 + 8;
+        bugframe = 4 * bugtype + 1;
+      }
+
+      ctx.clearRect(0, 0, canvas.width, canvas.height);
+
+      const bgtiles = isDarkmode() ? bgtilesw : bgtilesb;
+      for (let i = int(distance / 32), x = -distance % 32; x < 600; x += 32, i++) {
+        ctx.drawImage(bgtiles, rand(i) < 0.75 ? 0 : 32, 0, 32, 24, x, 150 - 40, 32, 24);
+      }
+      ctx.drawImage(bug, 40 * bugframe, 0, 40, 40, bugx, bugy, 40, 40);
+      ctx.drawImage(beetle, 40 * playerframe, 0, 40, 40, 0, 86 - playery, 40, 40);
+
+      let scorestr = String(score).padStart(5, "0");
+      if (highscore) scorestr = "HI " + String(highscore).padStart(5, "0") + " " + scorestr;
+      drawText(scorestr, 600 - scorestr.length * 8, 0, 1);
+      if (gamestate == STATE_GAME) {
+        if (score >= 100 && score < 8000 && score % 100 < 20 && score & 2) {
+          drawText("SPEED UP", 600 - 64, 12, 1);
+        }
+      } else if (gamestate == STATE_OVER) {
+        drawText("G A M E   O V E R", 300 - 17 * 8, 75 - 8, 2);
+      }
+    }
+
+    function onAction(e) {
+      e.preventDefault();
+      if (gamestate == STATE_INIT) {
+        gamestate = STATE_GAME;
+      } else if (gamestate == STATE_GAME) {
+        if (!jumping) {
+          jumping = true;
+          playerdy = 400;
+        }
+      } else if (gamestate == STATE_OVER) {
+        gamestate = STATE_INIT;
+        const score = int(distance / 10); 
+        if (score > highscore) highscore = score;
+        distance = 0;
+        jumping = false;
+        playery = 0;
+        playerdy = 0;
+      }
+    }
+
+    function drawText(text, x, y, s) {
+      const font = isDarkmode() ? fontw : fontb;
+      for (let i = 0; i < text.length; i++) {
+        const code = text.charCodeAt(i);
+        ctx.drawImage(font, code % 16 * 8, int(code / 16) * 8, 8, 8, x + s * 8 * i, y, s * 8, s * 8);
+      }
+    }
+
+    function overlap(b1, b2) {
+      return b1.x < b2.x + b2.w && b1.x + b1.w > b2.x && b1.y < b2.y + b2.h && b1.y + b1.h > b2.y;
+    }
+
+    function int(x) {
+      return Math.floor(x);
+    }
+
+    function hash32(i) {
+      return (i * 2654435761) >>> 0;
+    }
+
+    function rand(seed) {
+      return hash32(seed) / 0xffffffff;
+    }
+
+    function isDarkmode() {
+      return window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
+    }
+  </script>
+</body>
+
+</html>
```

Comment:
Add a newline, and, ideally, an automated check that we always have a newline for .html (docs.add sounds like a good place for the check?)